napari/layers/tracks/tracks.py

```diff
@@ -0,0 +1,435 @@
+# from napari.layers.base.base import Layer
+# from napari.utils.events import Event
+# from napari.utils.colormaps import AVAILABLE_COLORMAPS
+
+from typing import Dict, List, Union
+
+import numpy as np
+
+from ...utils.colormaps import AVAILABLE_COLORMAPS
+from ...utils.events import Event
+from ..base import Layer
+from ._track_utils import TrackManager
+
+
+class Tracks(Layer):
+    """ Tracks
+
+    A Tracks layer for overlaying trajectories on image data.
+
+    Parameters
+    ----------
+    data : array (N, D)
+        Coordinates for N points in D dimensions. T(Z)YX
+    properties : dict {str: array (N,)}, DataFrame
+        Properties for each point. Each property should be an array of length N,
+        where N is the number of points. Must contain 'track_id'
+    graph : dict {int: list}
+        Graph representing track edges. Dictionary defines the mapping between
+        a track ID and the parents of the track. This can be one (the track
+        has one parent, and the parent has >=1 child) in the case of track
+        splitting, or more than one (the track has multiple parents, but
+        only one child) in the case of track merging.
+    color_by: str
+        track property (from property keys) to color vertices by
+    edge_width : float
+        Width for all vectors in pixels.
+    tail_length : float
+        Length of the projection of time as a tail, in units of time.
+    colormap : str
+        Default colormap to use to set vertex colors. Specialized colormaps,
+        relating to specified properties can be passed to the layer via
+        colormaps_dict.
+    colomaps_dict : dict {str: Colormap}
+        dictionary list of colormap objects to use for coloring by track
+        properties.
+    name : str
+        Name of the layer.
+    metadata : dict
+        Layer metadata.
+    scale : tuple of float
+        Scale factors for the layer.
+    translate : tuple of float
+        Translation values for the layer.
+    opacity : float
+        Opacity of the layer visual, between 0.0 and 1.0.
+    blending : str
+        One of a list of preset blending modes that determines how RGB and
+        alpha values of the layer visual get mixed. Allowed values are
+        {'opaque', 'translucent', and 'additive'}.
+    visible : bool
+        Whether the layer visual is currently being displayed.
+
+    """
+
+    # The max number of points that will ever be used to render the thumbnail
+    # If more points are present then they are randomly subsampled
+    _max_tracks_thumbnail = 1024
+
+    def __init__(
+        self,
+        data,
+        *,
+        properties=None,
+        graph=None,
+        edge_width=2,
```

Comment:
I think this is called `tail width` in the layer controls. I think it would be good to unify the naming.